### PR TITLE
Headless Unit Tests

### PR DESCRIPTION
**Pull request by whiskey**
## Description
To handle some issues with tests this PR addresses several patterns within the app:

1. it removes the AppDelegate from the test target and thus reduces 'noise' during unit tests
1. as a consequence, some parts are moved out of the AppDelegate to their own module (see `AppDelegate & Globals` in the project outliner)
1. it increases the test timeouts to 5, 10, 20, 30 seconds for the `.short.`, etc. constants

## Link to Jira
n/a

## Screenshots
You don't want to see pictures with these keywords…!

cartoon-chicken-giving-thumbs-vector-260nw-663031879

### CODE REVIEW

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4958,7 +4970,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 12.5;
```
Comment (marcussc) [inline src/xcode/ENA/ENA.xcodeproj/project.pbxproj:4961]: Please don't :D

Comment (whiskey) [reply src/xcode/ENA/ENA.xcodeproj/project.pbxproj:4961]: Sorry, had that thing in mind the last days… Will be out/in/back/$whatever in a moment

File: src/xcode/ENA/TestPlans/AllTests.xctestplan [hunk 2]
```diff
@@ -72,6 +72,9 @@
   "testTargets" : [
     {
       "parallelizable" : true,
+      "skippedTests" : [
+        "StatisticsProviderTests\/testFetchLiveStats()"
```
Comment (pascal-brause) [inline src/xcode/ENA/TestPlans/AllTests.xctestplan:76]: why is this test skipped?

**Review by pascal-brause — APPROVED** [review]

**Review by tcfos — APPROVED** [review]

File: src/xcode/ENA/TestPlans/AllTests.xctestplan (hunk 2 above)
Comment (marcussc) [reply src/xcode/ENA/TestPlans/AllTests.xctestplan:76]: Backend didn't deliver so test would fail…

**Pull request merged**